.github/workflows/docs-preview-cleanup.yml

```diff
@@ -0,0 +1,32 @@
+name: docs-preview-cleanup
+
+permissions:
+  contents: write
+
+on:
+  # This always uses the workflow file from the base branch, and only runs if
+  # base == main
+  pull_request_target:
+    types: [closed]
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+
+      - run: mkdir -p empty-dir
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@62fec3add6773ec5dbbf18d2ee4260911aa35cf4
+        with:
+          folder: empty-dir
```

Comment:
(Thinking out loud, we can fiddle with this later if necessary) This might create a lot of empty dirs.

I don't think it matters much, but perhaps a better approach would be:

- To add a file called `PR_CLOSED` to the dir here
- Have an action in the docs repo that deletes all dirs with PR_CLOSED marker if it's over 7 days old.

I'm not sure how well that will work if that action happens to happen while another action is deploying to the docs. But I'm also not sure it'd be worse than with this PR as is; I'm not sure how this PR will do if the cleanup action on one PR is running at the same time as the deploy action acts on another PR.

It is something we can discover experimentally, though.

Reply:
> This might create a lot of empty dirs.

It actually deletes the dir entirely (I checked).

> I'm not sure how well that will work if that action happens to happen while another action is deploying to the docs.

Good catch. In fact, GHA doesn't support a "workflow queue" at all — if you use `concurrency:`, it will only keep one pending job in the queue.

I will fix that by disabling `force` and `single-commit` and switching to per-PR concurrency groups.

Reply:
Update: I will also delete the cleanup entirely b/c the extra complexity is not worth it

Reply:
(Thinking out loud) Would just adding the `PR_CLOSED` file (or something like that) be too complex as well? Then we could occasionally clean up manually.

Unless we nuke the repo history, garbage collection won't save much git storage, but it might save on the amount of traffic needed to deploy the website. OTOH, GitHub seems to take care of that, so maybe it's not a problem (or we could assume that they are smart about it and use `rsync`).

Ultimately, this is up to you.